grails-app/controllers/org/pih/warehouse/api/LocalizationApiController.groovy

```diff
@@ -21,30 +21,41 @@ class LocalizationApiController {
     GrailsApplication grailsApplication
 
     def list() {
-        String languageCode = params.lang
+        String languageCode = params.languageCode
         String prefix = params.prefix
 
         String[] supportedLocales = grailsApplication.config.openboxes.locale.supportedLocales
 
+        Locale defaultLocale = Locale.default
         Locale currentLocale = localizationService.getCurrentLocale()
-        Locale defaultLocale = currentLocale.getDefault()
-        Locale selectedLocale = languageCode ? LocalizationUtil.getLocale(languageCode) : currentLocale
+        Locale selectedLocale = languageCode ? StringUtils.parseLocaleString(languageCode) : currentLocale
 
-        // Get the default message properties as well as the message properties for the selected locale
+        // Get the system default message properties
         Properties defaultMessageProperties = localizationService.getMessagesProperties(defaultLocale)
+
+        // Get the fallback language default message properties if it is country specific language
+        Properties fallbackMessageProperties = null
+        if (selectedLocale.country && supportedLocales.contains(selectedLocale.language)) {
```

Comment:
I assume the only Locale for now, that would have the `country` not `null` is the `es-MX`, and this is the case for checking the `selectedLocale.country`?
Maybe it's worth having a comment about that (just a suggestion, because I needed a few seconds to figure out that part)

Reply:
Sure I can improve the comment a little bit more